### PR TITLE
Set foundation for introducing integration tests

### PR DESCRIPTION
**Pull request by geoand**
The commit creates the necessary Openshift + Minimal Istio cluster
needed to test against an Istio cluster that verifies the received
resources.
A couple of integration tests are introduced as part of this PR in order
with intent of being a guide for later introducing tests for other
Istio resources as well

### CODE REVIEW

File: istio-client/src/test/java/me/snowdrop/istio/client/it/GatewayIT.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  @Test
+  public void checkBasicGateway() {
+    //given
+    final IstioResource gateway = new IstioResourceBuilder()
+        .withApiVersion("networking.istio.io/v1alpha3")
+        .withNewMetadata()
+        .withName("httpbin-gateway")
+        .endMetadata()
+        .withNewGatewaySpec()
+        .withSelector(new HashMap<String, String>() {{
```
Comment (metacosm) [inline istio-client/src/test/java/me/snowdrop/istio/client/it/GatewayIT.java]: Use `addToSelector("istio", "ingressgateway")` instead.

Comment (geoand) [reply istio-client/src/test/java/me/snowdrop/istio/client/it/GatewayIT.java]: Thanks, done

**Comment by metacosm** [comment]
Thank you!

**Comment by geoand** [comment]
Sure thing :)

**Pull request merged**